ui/src/dialogs/EntitySetDeleteDialog/EntitySetDeleteDialog.jsx

```diff
@@ -30,51 +22,118 @@ class EntitySetDeleteDialog extends Component {
     this.onDelete = this.onDelete.bind(this);
   }
 
-  onDelete() {
-    const { entitySet, navigate, intl } = this.props;
-    this.props
-      .deleteEntitySet(entitySet.id)
-      .then(() =>
-        showSuccessToast(
-          intl.formatMessage(messages.success, { title: entitySet.label })
-        )
-      );
+  async onDelete() {
+    const { entitySet, navigate, intl, deleteEntitySet } = this.props;
+    const successMessage = intl.formatMessage(messages.success, {
+      title: entitySet.label,
+    });
 
+    await deleteEntitySet(entitySet.id);
+    showSuccessToast(successMessage);
     navigate(getCollectionLink({ collection: entitySet.collection }));
   }
 
   render() {
-    const { intl, entitySet } = this.props;
+    const { entitySet } = this.props;
     const { label, type } = entitySet;
 
+    const title = (
+      <>
+        {type === 'diagram' && (
+          <FormattedMessage
+            id="entityset.delete.title.diagram"
+            defaultMessage="Delete network diagram"
+            values={{ type }}
+          />
+        )}
+        {type === 'timeline' && (
+          <FormattedMessage
+            id="entityset.delete.title.timeline"
+            defaultMessage="Delete timeline"
+            values={{ type }}
+          />
+        )}
+        {type === 'list' && (
+          <FormattedMessage
+            id="entityset.delete.title.list"
+            defaultMessage="Delete list"
+            values={{ type }}
+          />
+        )}
+        {type === 'profile' && (
+          <FormattedMessage
+            id="entityset.delete.title.profile"
+            defaultMessage="Delete profile"
+            values={{ type }}
+          />
+        )}
+      </>
+    );
+
+    const buttonLabel = (
+      <>
+        <FormattedMessage
+          id="entityset.delete.confirm"
+          defaultMessage="I understand the consequences."
+          values={{ type }}
+        />{' '}
+        {type === 'diagram' && (
+          <FormattedMessage
+            id="entityset.delete.confirm.diagram"
+            defaultMessage="Delete this network diagram."
+            values={{ type }}
+          />
+        )}
+        {type === 'timeline' && (
+          <FormattedMessage
+            id="entityset.delete.confirm.timeline"
+            defaultMessage="Delete this timeline."
+            values={{ type }}
+          />
+        )}
+        {type === 'list' && (
+          <FormattedMessage
+            id="entityset.delete.confirm.list"
+            defaultMessage="Delete this list."
+            values={{ type }}
+          />
+        )}
+        {type === 'profile' && (
+          <FormattedMessage
+            id="entityset.delete.confirm.profile"
+            defaultMessage="Delete this profile."
+            values={{ type }}
+          />
+        )}
+      </>
```

Comment:
Nitpick. To me it would be better to create a set of values and then make only one <FormattedMessage /> element.